core/dbt/config/catalogs.py

```diff
@@ -0,0 +1,88 @@
+import os
+from typing import Any, Dict, List, Optional
+
+from dbt.artifacts.resources import Catalog, CoreCatalogIntegrationConfig
+from dbt.clients.yaml_helper import load_yaml_text
+from dbt.config.renderer import SecretRenderer
+from dbt.constants import CATALOGS_FILE_NAME
+from dbt.exceptions import YamlLoadError
+from dbt_common.clients.system import load_file_contents
+from dbt_common.exceptions import CompilationError, DbtValidationError
+
+
+def read_catalogs(project_dir: str, project_name: str) -> Dict[str, Any]:
+    path = os.path.join(project_dir, CATALOGS_FILE_NAME)
+
+    if os.path.isfile(path):
+        try:
+            contents = load_file_contents(path, strip=False)
+            yaml_content = load_yaml_text(contents)
+
+            if not yaml_content:
+                raise DbtValidationError(f"The file at {path} is empty")
+
+            return yaml_content
+        except DbtValidationError as e:
+            raise YamlLoadError(project_name=project_name, path=CATALOGS_FILE_NAME, exc=e)
+
+    return {}
+
+
+def load_single_catalog(raw_catalog: Dict[str, Any], renderer: SecretRenderer) -> Catalog:
```

Comment:
Will this handle getting the built-in/default integration for Snowflake? We need to add an integration for the user to be backwards compatible with the existing Iceberg implementation. Even without solving for backwards compatibility, we should add the default catalog for the user. This means the adapter will add an integration that `dbt-core` will not know about because it's not in the user's config.

Reply:
I think so? 
In the tests, `dbt-core` has no idea about `TestCatalogIntegration`; it is coming purely from the adapter.